checks/evaluation/license.go

```diff
@@ -15,81 +15,91 @@
-		Type:   finding.FileTypeNone,
-		Text:   "",
-		Offset: 1,
-	}
-	msg.Path = f.File.Path
-	msg.Type = finding.FileTypeSource
-	// #1 a license file was found.
-	score += 6
-
-	// #2 the licence was found at the top-level or LICENSE/ folder.
-	switch f.LicenseInformation.Attribution {
-	case checker.LicenseAttributionTypeAPI, checker.LicenseAttributionTypeHeuristics:
-		// both repoAPI and scorecard (not using the API) follow checks.md
-		// for a file to be found it must have been in the correct location
-		// award location points.
-		score += 3
-		msg.Text = "License file found in expected location"
-		dl.Info(&msg)
-		// for repo attribution prepare warning if not an recognized license"
-		msg.Text = "Any licence detected not an FSF or OSI recognized license"
-	case checker.LicenseAttributionTypeOther:
-		// TODO ascertain location found
-		score += 0
-		msg.Text = "License file found in unexpected location"
-		dl.Warn(&msg)
-		// for non repo attribution not the license detection is not supported
-		msg.Text = "Detecting license content not supported"
-	default:
-	}
-
-	// #3 is the license either an FSF or OSI recognized/approved license
-	if f.LicenseInformation.Approved {
-		score += 1
-		msg.Text = "FSF or OSI recognized license"
-		dl.Info(&msg)
-	} else {
-		// message text for this condition set above
-		dl.Warn(&msg)
+) checker.CheckResult {
+	// We have 3 unique probes, each should have a finding.
+	expectedProbes := []string{
+		hasLicenseFile.Probe,
+		hasApprovedLicense.Probe,
+		hasLicenseFileAtTopDir.Probe,
 	}
-	return score
-}
 
-// License applies the score policy for the License check.
-func License(name string, dl checker.DetailLogger,
-	r *checker.LicenseData,
-) checker.CheckResult {
-	var score int
-	if r == nil {
-		e := sce.WithMessage(sce.ErrScorecardInternal, "empty raw data")
+	if !finding.UniqueProbesEqual(findings, expectedProbes) {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "invalid probe results")
 		return checker.CreateRuntimeErrorResult(name, e)
 	}
 
-	// Apply the policy evaluation.
-	if r.LicenseFiles == nil || len(r.LicenseFiles) == 0 {
-		return checker.CreateMinScoreResult(name, "license file not detected")
+	// Compute the score.
+	score := 0
+	m := make(map[string]bool)
+	for i := range findings {
+		f := &findings[i]
+		if f.Outcome == finding.OutcomePositive {
+			switch f.Probe {
+			case hasApprovedLicense.Probe:
+				dl.Info(&checker.LogMessage{
+					Type:   finding.FileTypeSource,
+					Offset: 1,
+					Path:   f.Message,
+					Text:   "FSF or OSI recognized license",
+				})
+				score += scoreProbeOnce(f.Probe, m, 1)
+			case hasLicenseFileAtTopDir.Probe:
+				dl.Info(&checker.LogMessage{
+					Type:   finding.FileTypeSource,
+					Offset: 1,
+					Path:   f.Message,
+					Text:   "License file found in expected location",
+				})
+				score += scoreProbeOnce(f.Probe, m, 3)
+			case hasLicenseFile.Probe:
+				score += scoreProbeOnce(f.Probe, m, 6)
+				m[f.Probe] = true
```

Comment:
this map manipulation is redundant, `scoreProbeOnce` does it already?

Reply:
Removed in https://github.com/ossf/scorecard/pull/3465/commits/6d23ef52a18155eb60597f21c2538d5ac55761d0